src/runtime/mod.rs

```diff
@@ -105,9 +90,17 @@ impl Runtime {
             }
         };
 
-        tokio::pin!(future);
+        local.spawn_local(drive);
```

Comment:
Here's the big change. Spawning the driver block at creation, rather than per block_on call.

Interesting. Here we let driver.tick get called whenever the fd becomes ready, even if there is no block_on happening.

It doesn't hurt anything that the complete calls the tick can invoke may try to call an op future's waker?

Oh ... I bet the LocalSet is only live, during the rt.block_on call. And when the block_on is done, the LocalSet goes quiescent again until/if another rt.block_on call is made.

Very interesting indeed.

It's nice how futures are designed to just be dropped, with no callbacks needing to be executed. I guess a future can always have its own drop method defined for cleanup if necessary.

Reply:
Yup, the `LocalSet` only runs when run during the block_on call. It runs on the current thread, not in a background one.